pkg/scheduler/framework/plugins/dynamicresources/extended/util.go

```diff
@@ -17,22 +17,51 @@ limitations under the License.
 package extended
 
 import (
+	"context"
+	"sort"
+
 	v1 "k8s.io/api/core/v1"
 	resourceapi "k8s.io/api/resource/v1"
+	"k8s.io/klog/v2"
 	fwk "k8s.io/kube-scheduler/framework"
 )
 
+// DeviceClassMapping creates the mapping of extended resource name to device class name.
+// It always includes the implicit extended resource name for each device class.
 func DeviceClassMapping(draManager fwk.SharedDRAManager) (map[v1.ResourceName]string, error) {
 	classes, err := draManager.DeviceClasses().List()
-	extendedResources := make(map[v1.ResourceName]string, len(classes))
```

Comment:
I'd propose to define this mapping as `map[v1.ResourceName]*resourceapi.DeviceClass` as it may be needed to access other fields in some usage scenarios. Current implementation wouldn't allow that and we'll have to query DeviceClass object from the API server.

Reply:
storing only the pointer is error prone, the caller of the function MUST NOT modify the object in that case. That said, I added another func to return the map of resource name to device class object as you suggested. It is not exported for now.

Reply:
it's ok to pass around in one package, caution must be taken to pass it across packages.

Reply:
Passing pointers across packages is exactly what happens all the time.

Reply:
made changes as suggested. PTAL